andela/ah-kgl-avengers-backend
# #165020214 user notifications

## What does this PR do?
user notification
## Description of Task to be completed?
- register followers as subscribers to that author
- registers users who reacted on an article as subscribers to that article.
- send email notification to author's subscribers if author posts a new article
- send email notification to article's subscribers if there is a new comment.
## How should this be manually tested?
1. Run these commands
     - clone https://github.com/andela/ah-kgl-avengers-backend.git
     - git checkout ft-user-notifications-165020214
     - npm install
     - npm test
2. Postman
     - create 2 users by sending a post request on  localhost:3000/api/v1/auth/signup
    -  check email and activate users
    - login user 1 by sending a post request on localhost:3000/api/v1/auth/login
    -  follow user 2 by sending a post request on localhost:3000/api/v1/profiles/:username/follow
    - login user 2 by sending post request on localhost:3000/api/v1/auth/login
    - write an article using post request on localhost:3000/api/v1/articles
    - user1 will receive a notification email 
## Any background context you want to provide?

## What are the relevant pivotal tracker stories?
[#165020214 user notification](https://www.pivotaltracker.com/story/show/165020214)
## Screenshots (if appropriate)
N/A
## Questions:
N/A
